### PR TITLE
Remove mapbox Point dependency from map-interactions. 

### PR DESCRIPTION
**Pull request by ibgreen**
Update eslint settings.
Use static for propTypes (now that it static members is included in stage-2).

### CODE REVIEW

File: src/map-interactions.react.js [hunk 1]
```diff
@@ -206,6 +214,7 @@ export default class MapInteractions extends Component {
   /* eslint-disable complexity, max-statements */
   @autobind
   _onWheel(event) {
+    event.stopPropagation();
```
Comment (abmai) [inline src/map-interactions.react.js]: We had an issue and a subsequent PR to remove this `stopPropagation` a couple of weeks ago. Is there a reason why propagation needs to be stopped here?

File: src/map.react.js [hunk 2]
```diff
@@ -278,10 +282,6 @@ export default class MapGL extends Component {
     return 'inherit';
   }
 
-  _getMap() {
```
Comment (abmai) [inline src/map.react.js]: Many applications have come to depend on this to get to the internal map instance. We should add a console warning to this for a couple of versions before deprecating it completely. Can we add this back with a console warning? Something like "you can access the `_map` object directly and this method will eventually be deprecated" will do.

File: src/map.react.js [hunk 3]
```diff
@@ -505,21 +504,33 @@ export default class MapGL extends Component {
     this._onMouseDown({pos});
   }
 
+  @autobind _onTouchDrag({pos}) {
```
Comment (abmai) [inline src/map.react.js]: Do we need to use de-structuring here vs. just passing the entire argument as `opt` through?

File: src/map.react.js [hunk 4]
```diff
@@ -505,21 +504,33 @@ export default class MapGL extends Component {
...
+    this._onMouseDrag({pos});
+  }
+
+  @autobind _onTouchRotate({pos, startPos}) {
```
Comment (abmai) [inline src/map.react.js]: Ditto as above.

File: src/map-interactions.react.js (hunk 1 above)
Comment (ibgreen) [inline src/map-interactions.react.js]: > Is there a reason why propagation needs to be stopped here?
Good catch - removed.

File: src/map.react.js (hunk 3 above)
Comment (ibgreen) [inline src/map.react.js]: > Do we need to use de-structuring here vs. just passing the entire argument as opt through?

Updated all the touch events to just pass on `opt`

File: src/map-interactions.react.js (hunk 1 above)
Comment (Apercu) [inline src/map-interactions.react.js]: https://github.com/uber/react-map-gl/pull/158 for reference

File: src/map.react.js (hunk 2 above)
Comment (abmai) [inline src/map.react.js]: Yep that sounds good. We can move towards removing `_getMap()` eventually as well but we should add deprecation warnings for a couple of versions so people aren't caught completely off guard.

**Comment by abmai** [comment]
Are the changes already made? I don't see them updated in the PR - sorry for the confusion, asking since your comments indicate that you already made the changes.

**Comment by ibgreen** [comment]
> @ibgreen Are the changes already made? I don't see them updated in 

Now they are. Was responding while I was editing the code, sorry for the confusion.

**Comment by abmai** [comment]
lgtm, merging

**Pull request merged**